Release builds busted

It looks like something about the wrappers extensions.  Then generate a pile of linker errors at the moment.  Debug works.

Linking SwiftProtobufTestSuite_iOS

```
Undefined symbols for architecture i386:
  "(extension in SwiftProtobuf):SwiftProtobuf.ProtobufWrapper.init (A.WrappedType.BaseType) -> A", referenced from:
      SwiftProtobufTestSuite_iOS.Test_Api.testInitializer () -> () in Test_Api.o
      SwiftProtobufTestSuite_iOS.Test_Conformance.(testRepeatedBoolWrapper () -> ()).(closure #1) in Test_Conformance.o
      SwiftProtobufTestSuite_iOS.Test_Conformance.(testRepeatedBoolWrapper () -> ()).(closure #2) in Test_Conformance.o
      SwiftProtobufTestSuite_iOS.Test_Wrappers.testDoubleValue () throws -> () in Test_Wrappers.o
      SwiftProtobufTestSuite_iOS.Test_Wrappers.testFloatValue () throws -> () in Test_Wrappers.o
      SwiftProtobufTestSuite_iOS.Test_Wrappers.(testStringValue () throws -> ()).(implicit closure #11) in Test_Wrappers.o
      SwiftProtobufTestSuite_iOS.Test_Wrappers.(testStringValue () throws -> ()).(implicit closure #13) in Test_Wrappers.o
      ...
  "protocol witness table for SwiftProtobuf.Google_Protobuf_BoolValue : SwiftProtobuf.ProtobufWrapper in SwiftProtobuf", referenced from:
      SwiftProtobufTestSuite_iOS.Test_Conformance.(testRepeatedBoolWrapper () -> ()).(closure #1) in Test_Conformance.o
      SwiftProtobufTestSuite_iOS.Test_Conformance.(testRepeatedBoolWrapper () -> ()).(closure #2) in Test_Conformance.o
  "protocol witness table for SwiftProtobuf.Google_Protobuf_FloatValue : SwiftProtobuf.ProtobufWrapper in SwiftProtobuf", referenced from:
      SwiftProtobufTestSuite_iOS.Test_Wrappers.testFloatValue () throws -> () in Test_Wrappers.o
  "protocol witness table for SwiftProtobuf.Google_Protobuf_DoubleValue : SwiftProtobuf.ProtobufWrapper in SwiftProtobuf", referenced from:
      SwiftProtobufTestSuite_iOS.Test_Wrappers.testDoubleValue () throws -> () in Test_Wrappers.o
  "protocol witness table for SwiftProtobuf.Google_Protobuf_StringValue : SwiftProtobuf.ProtobufWrapper in SwiftProtobuf", referenced from:
      SwiftProtobufTestSuite_iOS.Test_Api.testInitializer () -> () in Test_Api.o
      SwiftProtobufTestSuite_iOS.Test_Wrappers.(testStringValue () throws -> ()).(implicit closure #11) in Test_Wrappers.o
      SwiftProtobufTestSuite_iOS.Test_Wrappers.(testStringValue () throws -> ()).(implicit closure #13) in Test_Wrappers.o
      SwiftProtobufTestSuite_iOS.Test_Api.testInitializer () -> () in Test_Api.o
      SwiftProtobufTestSuite_iOS.Test_Wrappers.(testStringValue () throws -> ()).(implicit closure #11) in Test_Wrappers.o
      SwiftProtobufTestSuite_iOS.Test_Wrappers.(testStringValue () throws -> ()).(implicit closure #13) in Test_Wrappers.o
ld: symbol(s) not found for architecture i386
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

--- a/Sources/SwiftProtobuf/Google_Protobuf_Wrappers+Extensions.swift
+++ b/Sources/SwiftProtobuf/Google_Protobuf_Wrappers+Extensions.swift
@@ -53,6 +53,12 @@ protocol ProtobufWrapper {
 }
 
 extension ProtobufWrapper {
+  // NOTE: The `init(_ value: WrappedType.BaseType)` initializer repeated below
+  // should theoretically be able to go here and be declared public, but this
+  // causes linker errors in release builds (see issue #70). If this is indeed a
+  // bug and should be allowed, we should move the initializer back into this
+  // extension once it's fixed, to reduce a small amount of code duplication/
+  // bloat.
 
   func serializeWrapperJSON() throws -> String {
     if !isZeroOrEmpty {

--- a/Sources/SwiftProtobuf/Google_Protobuf_Wrappers+Extensions.swift
+++ b/Sources/SwiftProtobuf/Google_Protobuf_Wrappers+Extensions.swift
@@ -54,11 +54,6 @@ protocol ProtobufWrapper {
 
 extension ProtobufWrapper {
 
-  public init(_ value: WrappedType.BaseType) {
-    self.init()
-    self.value = value
-  }
-
   func serializeWrapperJSON() throws -> String {
     if !isZeroOrEmpty {
       var encoder = ProtobufJSONEncoder()
@@ -78,6 +73,11 @@ extension Google_Protobuf_DoubleValue:
 
   var isZeroOrEmpty: Bool {
     return value == 0
+  }
+
+  public init(_ value: WrappedType.BaseType) {
+    self.init()
+    self.value = value
   }
 
   public init(floatLiteral: FloatLiteralType) {
@@ -107,6 +107,11 @@ extension Google_Protobuf_FloatValue:
     return value.isZero
   }
 
+  public init(_ value: WrappedType.BaseType) {
+    self.init()
+    self.value = value
+  }
+
   public init(floatLiteral: FloatLiteralType) {
     self.init(floatLiteral)
   }
@@ -132,6 +137,11 @@ extension Google_Protobuf_Int64Value:
 
   var isZeroOrEmpty: Bool {
     return value == 0
+  }
+
+  public init(_ value: WrappedType.BaseType) {
+    self.init()
+    self.value = value
   }
 
   public init(integerLiteral: IntegerLiteralType) {
@@ -161,6 +171,11 @@ extension Google_Protobuf_UInt64Value:
     return value == 0
   }
 
+  public init(_ value: WrappedType.BaseType) {
+    self.init()
+    self.value = value
+  }
+
   public init(integerLiteral: IntegerLiteralType) {
     self.init(integerLiteral)
   }
@@ -186,6 +201,11 @@ extension Google_Protobuf_Int32Value:
 
   var isZeroOrEmpty: Bool {
     return value == 0
+  }
+
+  public init(_ value: WrappedType.BaseType) {
+    self.init()
+    self.value = value
   }
 
   public init(integerLiteral: IntegerLiteralType) {
@@ -215,6 +235,11 @@ extension Google_Protobuf_UInt32Value:
     return value == 0
   }
 
+  public init(_ value: WrappedType.BaseType) {
+    self.init()
+    self.value = value
+  }
+
   public init(integerLiteral: IntegerLiteralType) {
     self.init(integerLiteral)
   }
@@ -240,6 +265,11 @@ extension Google_Protobuf_BoolValue:
 
   var isZeroOrEmpty: Bool {
     return !value
+  }
+
+  public init(_ value: WrappedType.BaseType) {
+    self.init()
+    self.value = value
   }
 
   public init(booleanLiteral: Bool) {
@@ -269,6 +299,11 @@ extension Google_Protobuf_StringValue:
 
   var isZeroOrEmpty: Bool {
     return value.isEmpty
+  }
+
+  public init(_ value: WrappedType.BaseType) {
+    self.init()
+    self.value = value
   }
 
   public init(stringLiteral: String) {
@@ -302,6 +337,11 @@ extension Google_Protobuf_BytesValue: ProtobufWrapper {
 
   var isZeroOrEmpty: Bool {
     return value.isEmpty
+  }
+
+  public init(_ value: WrappedType.BaseType) {
+    self.init()
+    self.value = value
   }
 
   public func serializeJSON() throws -> String {

